lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2753,27 +2820,31 @@ void addAnnotationDecorations(SPIRVEntry *E, DecorationsInfoVec &Decorations) {
 
     switch (I.first) {
     case DecorationUserSemantic:
-      E->addDecorate(new SPIRVDecorateUserSemanticAttr(E, I.second));
+      assert(I.second.size() == 1);
+      E->addDecorate(new SPIRVDecorateUserSemanticAttr(E, I.second[0]));
       break;
     case DecorationMemoryINTEL: {
       if (M->isAllowedToUseExtension(
-              ExtensionID::SPV_INTEL_fpga_memory_attributes))
-        E->addDecorate(new SPIRVDecorateMemoryINTELAttr(E, I.second));
+              ExtensionID::SPV_INTEL_fpga_memory_attributes)) {
+        assert(I.second.size() == 1);
+        E->addDecorate(new SPIRVDecorateMemoryINTELAttr(E, I.second[0]));
+      }
     } break;
     case DecorationMergeINTEL: {
       if (M->isAllowedToUseExtension(
               ExtensionID::SPV_INTEL_fpga_memory_attributes)) {
-        StringRef Name = StringRef(I.second).split(':').first;
-        StringRef Direction = StringRef(I.second).split(':').second;
+        assert(I.second.size() == 2);
         E->addDecorate(
-            new SPIRVDecorateMergeINTELAttr(E, Name.str(), Direction.str()));
+            new SPIRVDecorateMergeINTELAttr(E, I.second[0], I.second[1]));
```

Comment:
Nice to mention what are `I.second[N]` (to add a comment that these are Name and Direction)

Reply:
Yes, please add the same comment right here too

Reply:
I thought I added a comment here, but it seems like it got lost somehow. It should be there now.